db-tests/src/test/java/gov/nasa/jpl/aerie/database/PlanMigrationTests.java

```diff
@@ -0,0 +1,356 @@
+          (result -> 'modified_activity_types')::text as modified_activity_types,
+          (result -> 'impacted_directives')::text as impacted_directives
+          from hasura.check_model_compatibility_for_plan(%d, %d)
+          """.formatted(planId, newModelId)
+      );
+
+      if (res.next()) {
+        final var resultMap = new HashMap<String, String>();
+        resultMap.put("removed_activity_types", res.getString("removed_activity_types"));
+        resultMap.put("modified_activity_types", res.getString("modified_activity_types"));
+        resultMap.put("impacted_directives", res.getString("impacted_directives"));
+        return resultMap;
+      } else {
+        throw new SQLException("No result returned from check_model_compatibility");
+      }
+    }
+  }
+
+  private Boolean migratePlanToModel(final int planId, final int newModelId) throws SQLException{
+    try(final var statement = connection.createStatement()){
+      final var res = statement.executeQuery(
+          //language=sql
+          """
+          select hasura.migrate_plan_to_model(%d, %d, '%s'::json);
+          """.formatted(planId, newModelId, merlinHelper.admin.session())
+      );
+      return res.next();
+    }
+  }
+
+  private Integer getPlanModel(final int planId) throws SQLException {
+    try (final var statement = connection.createStatement()) {
+      final var res = statement.executeQuery(
+          //language=sql
+          """
+          select model_id
+          from merlin.plan
+          where id = %d;
+          """.formatted(planId));
+      assertTrue(res.next());
+      return res.getInt(1);
+    }
+  }
+
+
+  /**
+   * Check that migration fails if:
+   *  (a) _plan_id is invalid and _new_model_id is valid
+   *  (b) _plan_id is valid and _new_model_id is invalid
+   *  (c) _plan_id and _new_model_id are both invalid
+   *
+   */
+  @Test
+  void checkFailsForNonexistentPlansOrModels() throws SQLException {
+    final var sqlEx = assertThrows(
+      SQLException.class,
+      () -> migratePlanToModel(-1, missionModelId)
+    );
+    assertTrue(sqlEx.getMessage().contains("Plan -1 does not exist"), "bad error message, got " + sqlEx.getMessage());
+
+
+    final var sqlEx2 = assertThrows(
+      SQLException.class,
+      () -> migratePlanToModel(planId, -1)
+    );
+    assertTrue(sqlEx2.getMessage().contains("Model -1 does not exist"), "bad error message, got " + sqlEx2.getMessage());
+
+
+    final var sqlEx3 = assertThrows(
+      SQLException.class,
+      () -> migratePlanToModel(-1, -1)
+    );
+    assertTrue(sqlEx3.getMessage().contains("Plan -1 does not exist"), "bad error message, got " + sqlEx3.getMessage());
+  }
+
+
+  /**
+   * Check that migration fails if there exist open merge requests on _plan_id.
+   */
+  @Test
+  void checkFailsForOpenMergeRequests() throws SQLException {
+    final int branchId = duplicatePlan(planId, "MultiTags Test");
+    final int mergeRQ = createMergeRequest(planId, branchId);
```

Comment:
The value of `mergeRQ` is unused and the assignment can be removed.

```suggestion
   createMergeRequest(planId, branchId);
```

Reply:
Done in 83096223a92a5f4f051e3a2804195e018703f48b